packages/kos-sdk/src/chains/tron/mod.rs

```diff
@@ -196,6 +200,62 @@ impl TRX {
         })
     }
 
+    async fn check_valid_address(
+        valid_address: &bool,
+        addr_sender: &address::Address,
+        addr_receiver: &address::Address,
+        amount: &BigNumber,
+        token: &str,
+        node: &str,
+    ) -> Result<kos_proto::tron::Transaction, Error> {
+        if !valid_address {
+            let contract = kos_proto::tron::TransferAssetContract {
+                owner_address: addr_sender.as_bytes().to_vec(),
+                to_address: addr_receiver.as_bytes().to_vec(),
+                amount: amount.to_i64(),
+                asset_name: token.as_bytes().to_vec(),
+            };
+            let transaction = requests::create_asset_transfer(node, contract).await?;
+            return Ok(transaction);
+        }
```

Comment:
Dont understand what you are doing here.... the function signature is `check_valid_address` but receive as param `valid_addres: bool` why a check if you already have defined by param?

also, what check valid address has to be with transfer asset?